src/tiled/wangdock.cpp

```diff
@@ -333,13 +431,31 @@ void WangDock::setCurrentWangSet(WangSet *wangSet)
     emit currentWangSetChanged(mCurrentWangSet);
 }
 
+void WangDock::activateErase()
+{
+    mEraseWangIdsButton->setChecked(true);
+
+    mCurrentWangId = 0;
+
+    mWangTemplateView->selectionModel()->clearCurrentIndex();
+    mWangTemplateView->selectionModel()->clearSelection();
+    mWangColorView->selectionModel()->clearCurrentIndex();
+    mWangColorView->selectionModel()->clearSelection();
+    emit currentWangIdChanged(mCurrentWangId);
+}
+
 void WangDock::retranslateUi()
 {
     setWindowTitle(tr("Wang Sets"));
 
     mEraseWangIdsButton->setText(tr("Erase WangIds"));
     mAddWangSet->setText(tr("Add Wang Set"));
     mRemoveWangSet->setText(tr("Remove Wang Set"));
+
+    if (mWangColorView->isVisible())
+        mSwitchTemplateViewButton->setText(tr("Switch to Template View"));
```

Comment:
I'll definitely save it for a later patch. As for the name change thats a good point, I'll think of something different.